### ISSUE TITLE
Comprehensive docs generation

### ISSUE THREAD

**Issue by ulrikstrid** [ocaml/dune#3436]
## Desired Behavior
I want to be able to build docs for my library and include some of the important dependencies in the generated output, at least in a way that I can link to their types.

I'm thinking of something like a flag that would look like this:
`dune build @doc --with-dependencies=lwt,httpaf,...`

Edit:
I think this is the same as or close to  #1949

**Comment by rgrinberg** [comment]
It's possible, but it's kind of a pain. I recall that @jonludlam went quite far into making it work, but I'm not sure if it was all worth it at the end. If Jon isn't working on it anymore, perhaps you could take a stab at it?

**Comment by ulrikstrid** [comment]
If I could get some pointers on where to start I could take a stab at it 👍

**Comment by jonludlam** [comment]
It would be amazing if you wanted to work on this!

I made a start on this with @rgrinberg a couple of months ago, but it's stalled - the code we worked on is here: https://github.com/jonludlam/dune/tree/odoc-external-libs

IIRC we were starting by making sure the modules section in the dune-package library stanzas were non-optional to ensure we've got a complete list of the modules associated with each library. Then we were going to compile the cmtis and output them into `_odoc/external/<lib name>` ready for linking. Happy to help if I can!
